x-pack/plugins/alerts/server/alerts_client.ts

```diff
@@ -525,27 +586,51 @@ export class AlertsClient {
   }
 
   public async muteAll({ id }: { id: string }) {
-    await this.savedObjectsClient.update('alert', id, {
+    const { attributes } = await this.unsecuredSavedObjectsClient.get<RawAlert>('alert', id);
```

Comment:
We shouldn't cache these anyway - role assignment changes, and this is checked not just per user but per operation made by a user so that you have an audit trail.
This aligns with how it's done for Spaces and the security team have confirmed this is both performant enough and the expected behaviour. 👍